source/Connection/AMDSClientTCPSocket.cpp

```diff
@@ -53,47 +53,52 @@ void AMDSClientTCPSocket::onSocketError(QAbstractSocket::SocketError error)
 
 void AMDSClientTCPSocket::readClientRequestMessage()
 {
+	QDataStream *inDataStream = 0;
+
 	if (tcpSocket_->bytesAvailable() < (int)sizeof(quint32))
 		return;
 
-	QDataStream *inDataStream = 0;
+	bool onResponseBoundary = false; // there might be multiple request in the buffer, we will read them out one by one
 	if (!waitingMorePackages_) {
+		readedBufferSize_ = 0;
+
 		inDataStream = new QDataStream(tcpSocket_);
```

Comment:
Since it is declared and 0'ed at the top, you could just do a check on valid pointer before calling new.

```c++
if (!inDataStream)
     inDataStream = new QDataStream(tcpSocket_);
```